app/fruits/[fruitName]/page.js

```diff
@@ -12,31 +14,53 @@ import Fruit from './Fruit';
 // ];
 
 export function generateMetadata({ params }) {
-  const singleFruit = fruits.find((fruit) => {
-    return fruit.name.toLowerCase() === params.fruitName;
-  });
+  const fruit = getFruitByName(params.fruitName);
 
-  if (!singleFruit) {
+  if (!fruit) {
     return rootNotFoundMetadata;
   }
 
   return {
-    title: singleFruit.name,
-    description: `Single fruit page for ${singleFruit.name}`,
+    title: fruit.name,
+    description: `Single fruit page for ${fruit.name}`,
   };
 }
 
 // we add this only if we have no dynamic function as cookies or headers
 export const dynamic = 'force-dynamic';
 
 export default function FruitPage({ params }) {
-  const singleFruit = fruits.find((fruit) => {
-    return fruit.name.toLowerCase() === params.fruitName;
-  });
+  const fruit = getFruitByName(params.fruitName);
 
-  if (!singleFruit) {
+  if (!fruit) {
     notFound();
   }
 
-  return <Fruit fruit={singleFruit} />;
+  const fruitComments = parseJson(getCookie('fruitComments'));
```

Comment:
in our code, I would probably go for a pattern more like this

but maybe for the lecture we can put this into two steps (2 variables) for the students, so that they understand what each line + each step does

let's do this for every usage of `parseJson()`

Reply:
863f5dd